pmd-core/src/main/java/net/sourceforge/pmd/Rule.java

```diff
@@ -28,13 +28,15 @@
      * The property descriptor to universally suppress violations with messages
      * matching a regular expression.
      */
+    // TODO 7.0.0 use PropertyDescriptor<Pattern>
     StringProperty VIOLATION_SUPPRESS_REGEX_DESCRIPTOR = new StringProperty("violationSuppressRegex",
```

Comment:
Ok, we can't switch it now, because the properties are a field of the interface Rule - which makes it public API.

Maybe we should remove it here in the (Java) API in 7.0.0?  The only API left would be, when using a rule and setting the properties in the ruleset xml. There the type doesn't matter - since the String is then automatically converted into a Pattern.

Reply:
Yeah it would make sense to hide the descriptors. But we should probably allow setting these from Java somehow, for external tools that edit rules. The descriptors are definitely an implementation detail